### PR TITLE
story: ESM with Typescript

### PR DESCRIPTION
**Pull request by burgerni10**

### CODE REVIEW

File: src/engine/base-engine.js [hunk 1]
```diff
@@ -69,6 +93,10 @@ class BaseEngine {
    * @returns {Promise<void>} - The result promise
    */
   async initEngineServices(engineConfig) {
+    // const packageJson = JSON.parse(await fs.readFile('package.json'))
+    // this.version = packageJson.version
+    // TODO
+    this.version = '2.4.0'
```
Comment (marouanehassanioptimistik) [inline src/engine/base-engine.js:99]: is this intended to be left and fixed in another PR?

File: src/index.js [hunk 2]
```diff
@@ -44,14 +42,17 @@ if (cluster.isMaster) {
   const baseDir = path.resolve(path.extname(configFile) ? path.parse(configFile).dir : configFile)
   process.chdir(baseDir)
   createFolder(baseDir).then(async () => {
+    // TODO
+    const packageJson = '2.4.0' // JSON.parse(await fs.readFile('package.json'))
```
Comment (marouanehassanioptimistik) [inline src/index.js:46]: same here

**Review by marouanehassanioptimistik — CHANGES_REQUESTED** [review]
the overall look seems ok i have seen some weird todos leftover are they fixed later?

File: src/engine/base-engine.js (hunk 1 above)
Comment (burgerni10) [reply src/engine/base-engine.js:99]: Fixed in another PR
